### PR TITLE
release pipeline related fixes

### PR DESCRIPTION
**Pull request by sfc-gh-pmardziel**
# Description

- Fixes a few things that got broken in the execution of old notebooks.

- Fixes a few tests that were run as part of the release pipeline.

  - Removed `database_file` arg as it was a deprecation warning for a long time.
  
  - Added more optional deps to the `tests-optional` group in `pyproject.toml` as they are required for the release tests.

  - Fix to `App.select_context` to be more informative in case optional modules are not installed.

  - Updated versions of various llama_index dependencies in `pyproject.toml` as I was finding bugs in older versions.

  - Small fix to `TruChain.main_input` for an edge/new case.

  - Typo fix in `n_cortext_guardrails_tokens`.

  - Added `Record.__str__` to print fewer, and more useful things than default `__str__` from pydantic: the calls and their paths.

  - Refactored `JSONTestCase` as I might need to extend other things than `unittest.TestCase` for async tests.

  - Temporarily disabled `deprecation.class_moved` mechanism due to interference with serialization. Also disabled the related unit test.

- TODO (will make a new PR for this) fix async handling. Started work on this but it is incomplete.

## Other details good to know for developers

Please include any other details of this change useful for _TruLens_ developers.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
  not work as expected)
- [ ] New Tests
- [ ] This change includes re-generated golden test results
- [ ] This change requires a documentation update

### CODE REVIEW

File: tests/test.py [hunk 1]
```diff
@@ -73,8 +96,8 @@ def module_installed(module: str) -> bool:
 T = TypeVar("T")
 
 
-def hashable_skip(obj: T, skips: Set[str]) -> T:
-    """Return a hashable copy of `obj` with all keys/attributes in `skips`
+def canonical(obj: T, skips: Set[str]) -> Union[T, Dict, Tuple]:
```
Comment (sfc-gh-jreini) [inline tests/test.py:99]: what is the significance of this change?

File: tests/test.py [hunk 2]
```diff
@@ -153,20 +181,20 @@ def str_sorted(seq: Sequence[T], skips: Set[str]) -> Sequence[T]:
...
-class JSONTestCase(TestCase):
+class WithJSONTestCase:
     """TestCase class that adds JSON comparisons and golden expectation handling."""
 
     def load_golden(self, path: Union[str, Path]) -> JSON:
         """Load the golden file `path` and return its contents.
 
         Args:
-            golden_filename: The name of the golden file to load. The file must
+            path: The name of the golden file to load. The file must
```
Comment (sfc-gh-jreini) [inline tests/test.py:197]: nit: golden_path?

File: tests/test.py (hunk 1 above)
Comment (sfc-gh-pmardziel) [reply tests/test.py:99]: Just the name, not using hashing any more so the name didn't make sense. Name now is more appropriate for what it does, create a "canonical" version of a JSON for sorting.

File: tests/test.py (hunk 2 above)
Comment (sfc-gh-pmardziel) [reply tests/test.py]: fixed

**Review by sfc-gh-jreini — APPROVED** [review]

**Pull request merged**